### ISSUE TITLE
phx.gen.live template code & tests no longer match up (specifically for .table core component)

### ISSUE THREAD

**Issue by benswift** [phoenixframework/phoenix#5294]
I'm sure this is just a "we'll get to it ASAP" thing, but just FYI the [table component in the index.html.heex](https://github.com/phoenixframework/phoenix/blob/master/priv/templates/phx.gen.live/index.html.heex#L10) doesn't provide a `@row_id` attribute (as per the [new table component in rc3](https://github.com/phoenixframework/phoenix/blob/master/installer/templates/phx_web/components/core_components.ex#L465)).

This works fine for clicking around, because `phx-click` knows what the right id is, but [the way the tests are set up](https://github.com/phoenixframework/phoenix/blob/master/priv/templates/phx.gen.live/live_test.exs#L52) uses the id as part of the selector to find the edit/delete buttons, so the tests fail.

Not sure exactly which way to fix this (either in the generated table, or in the genrated tests), but one way is just to add to the generated table a `@row_id` attribute (a function) which generates the correct ids:

```elixir
row_id={&"NAME-#{&1.id}"} ## using the EEx tempate engine to put the correct schema name in there
```

**Comment by josevalim** [comment]
@benswift the new index is using streams and the IDs are lifted from the streams themselves:

https://github.com/phoenixframework/phoenix/blob/master/installer/templates/phx_web/components/core_components.ex#L480-L482

:)

**Comment by benswift** [comment]
Ah, thanks @josevalim .

My (deeper) issue was that this isn't a `phx.new` for rc3, it was a recent rc2 that I decided to upgrade (mostly for staying-on-the-bleeding-edge giggles; it's not deployed anywhere just yet and I can afford things breaking sometimes). So I updated `core_components.ex`, and had noticed the change to the attrs expected by `<.table>`, but I hadn't noticed the change to streams in the controller module (so that `with` clause you linked wasn't matching).

I guess my question, then, is (given my explanation of the context above) should I switch things over to streams now? Or is it worth just using my "set `@row_id` explicitly" hack for now and waiting a bit until the streams stuff settles down a bit more?

Thanks for all your work btw - and I _hope_ I'm able to be helpful in testing stuff out on the bleeding edge rather than just providing noise while you're trying to get things done. Let me know if I've misjudged that 😉
